ports/esp32/boards/ESP32_GENERIC_S3/mpconfigboard.cmake

```diff
@@ -19,3 +19,14 @@ if(MICROPY_BOARD_VARIANT STREQUAL "SPIRAM_OCT")
         MICROPY_HW_BOARD_NAME="Generic ESP32S3 module with Octal-SPIRAM"
     )
 endif()
+
+if(MICROPY_BOARD_VARIANT STREQUAL "FLASHSIZE_4MB")
+    set(SDKCONFIG_DEFAULTS
+        ${SDKCONFIG_DEFAULTS}
+        boards/ESP32_GENERIC_S3/sdkconfig.flash_4mb
+    )
+
+    list(APPEND MICROPY_DEF_BOARD
+        MICROPY_HW_BOARD_NAME="Generic ESP32S3 module with 4MiB flash size"
```

Comment:
I don't think the board name needs to change.  The flash size doesn't really need to be advertised at the REPL banner.

Reply:
agreed, removed custom board name and renamed sdkconfig.flash_4mb to sdkconfig.flash_4m for consistency